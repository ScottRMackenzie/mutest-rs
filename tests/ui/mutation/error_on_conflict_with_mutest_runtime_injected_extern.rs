//@ build: fail
//@ stderr
//@ aux-build: phf.rs
//@ rustc-flags: --extern __mutest_runtime_public_dep_phf=target/mutest_test/debug/deps/auxiliary/libphf.rlib

// NOTE: A mutable function is required to generate conflicting phf-dependent harness statements.
fn mutable_add_fn(a: usize, b: usize) -> usize {
    a + b
}

#[test]
fn test() {
    __mutest_runtime_public_dep_phf::incompatible_phf_map!();

    // NOTE: Mutations are required to generate conflicting phf-dependent harness statements.
    assert_eq!(3, mutable_add_fn(1, 2));
}
