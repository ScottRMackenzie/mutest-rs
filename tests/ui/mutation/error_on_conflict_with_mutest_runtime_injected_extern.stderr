error: mutest-injected crate conflicts with existing extern `__mutest_runtime_public_dep_phf`
  |
  = note: mutest-injected crates use the reserved `__mutest_runtime_public_dep_` prefix: if you see this error for any other crate, please file a bug report

